command/agent/consul/client.go

```diff
@@ -867,6 +869,11 @@ func (c *ServiceClient) UpdateGroup(oldAlloc, newAlloc *structs.Allocation) erro
 		return fmt.Errorf("task group %q not in old allocation", oldAlloc.TaskGroup)
 	}
 
+	if len(oldTG.Services) == 0 {
+		// No old group services, simply add new group services
+		return c.RegisterGroup(newAlloc)
```

Comment:
This is necessary to prevent makeAllocTaskServices below from erroring on the old alloc due to the lack of a group network and causing the update hook to error.